homeassistant/components/device_tracker/legacy.py

```diff
@@ -609,7 +610,15 @@ def async_get_extra_attributes(self, device: str) -> Any:
         except vol.Invalid as exp:
             async_log_exception(exp, dev_id, devices, hass)
         else:
-            result.append(Device(hass, **device))
+            device_entity = Device(hass, **device)
+            now = dt_util.utcnow()
+            run = RecorderRuns(start=(now - timedelta(hours=24)))
+            last_state = history.get_state(hass, now, device_entity.entity_id, run)
```

Comment:
Legacy device tracker entities are already using the last saved state via the restore state feature.

https://github.com/home-assistant/home-assistant/blob/770ad86f126eb890f6c4021dc72261af62acd862/homeassistant/components/device_tracker/legacy.py#L507-L529

Reply:
but in my test, 
`state = await self.async_get_last_state()`
do not reload lastest device_tracker state from database

Reply:
It's reloaded from a json file.

Reply:
I use ios hass app to push device_tracker info.   it is worked ok in hass, but I can't see device_tracker info in json files (.storage/core.restore_state, core.device_registry ).

so I want to reload device_tracker state from database

Reply:
Here's the relevant code for the mobile app that supports the ios app:
https://github.com/home-assistant/home-assistant/blob/3efdf29dfa178838002201eaea258db78e6fa2ba/homeassistant/components/mobile_app/device_tracker.py#L115-L139

Reply:
Make sure you have the `mobile_app:` integration activated in your configuration.yaml.